sdks/python/gen_xlang_wrappers.py

```diff
@@ -0,0 +1,366 @@
+      default_service = f"BeamJarExpansionService(\"{default_service}\")"
+
+      # use jinja to generate the code
+      python_wrapper_class = python_wrapper_template.render(
+          class_name=name,
+          identifier=identifier,
+          parameters=parameters,
+          description=description,
+          default_expansion_service=default_service)
+
+      if destination not in destinations:
+        destinations[destination] = []
+      destinations[destination].append(python_wrapper_class)
+
+  return destinations
+
+
+def write_wrappers_to_destinations(grouped_wrappers: Dict[str, List[str]]):
+  """
+  Takes a dictionary of generated wrapper code, grouped by destination.
+  For each destination, create a new file containing the respective wrapper
+  classes. Each file includes the Apache License header and relevant imports.
+  Note: the Jinja template should already follow linting and formatting rules.
+  """
+
+  for dest, wrappers in grouped_wrappers.items():
+    dest += PYTHON_SUFFIX
+    with open(dest, "w") as file:
+      file.write(LICENSE_HEADER.lstrip())
+      file.write(
+          "# NOTE: This file contains autogenerated external transform(s)\n"
+          "# and should not be edited by hand.\n\n")
+      file.write("# pylint:disable=line-too-long\n\n")
+      file.write(
+          "from apache_beam.transforms.external import "
+          "BeamJarExpansionService\n"
+          "from apache_beam.transforms.external_schematransform_provider "
+          "import ExternalSchemaTransform\n")
+      for wrapper in wrappers:
+        file.write(wrapper + "\n")
+
+
+def delete_generated_files(root_dir):
+  """Scans for and deletes generated wrapper files."""
+  logging.info("Deleting external transform wrappers from dir %s", root_dir)
+  for path in find_by_ext(root_dir, PYTHON_SUFFIX):
+    os.remove(path)
+
+
+def run_script(cleanup, input_expansion_services, output_transforms_config):
+  # Cleanup first if requested. This is needed to remove outdated wrappers.
+  if cleanup:
+    delete_generated_files(os.path.join(PYTHON_SDK_ROOT, 'apache_beam'))
+
+  # Find paths for expansion service YAML source and
+  # transform config YAML output
+  expansion_services_source = os.path.join(
+      os.path.dirname(__file__), input_expansion_services)
+  transforms_config_file = os.path.join(
+      os.path.dirname(__file__), output_transforms_config)
+
+  # Generate transform configs and output to the specified path
+  generate_transform_configs(
+      input_services=expansion_services_source,
+      output_file=transforms_config_file)
+
+  # Build and get the generated wrapper code
+  grouped_wrappers = get_wrappers_from_transform_configs(transforms_config_file)
+
+  # Write generated code to the appropriate destinations
+  write_wrappers_to_destinations(grouped_wrappers)
+
+  # Finally, delete the transform YAML config as we don't need it anymore
+  os.remove(transforms_config_file)
+
+
+if __name__ == '__main__':
```

Comment:
How will we test the generated classes ? I guess we are adding ITs for known key generated transforms separately (or update existing tests) ?

Reply:
This PR includes tests to check that the script generates files properly and has one IT using the GenerateSequence transform. The generated classes are essentially thin wrappers of `ExternalSchemaTransform`, which has some more tests in external_schematransform_provider_test.py: https://github.com/apache/beam/blob/b5dc728b677101cf3968e9f94db0898342343f6e/sdks/python/apache_beam/transforms/external_schematransform_provider_test.py#L104